src/ripple/overlay/tests/manifest_test.cpp

```diff
@@ -0,0 +1,102 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright 2014 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/basics/TestSuite.h>
+#include <ripple/overlay/impl/Manifest.h>
+#include <ripple/protocol/Sign.h>
+#include <ripple/protocol/STExchange.h>
+
+namespace ripple {
+namespace tests {
+
+class manifest_test : public ripple::TestSuite
+{
+public:
+    // Return a manifest in both serialized and STObject form
+    std::string
+    make_manifest(AnySecretKey const& sk, AnyPublicKey const& spk, int seq)
+    {
+        auto const pk = sk.publicKey();
+        
+        STObject st;
+        set(st, sfSequence, seq);
+        set(st, sfPublicKey, pk);
+        set(st, sfSigningPubKey, spk);
+
+        sign(st, HashPrefix::manifest, sk);
+        expect(verify(st, HashPrefix::manifest, pk));
+
+        Serializer s;
+        st.add(s);
+
+        std::string const m (static_cast<char const*> (s.data()), s.size());
+        return m;
+    }
+
+    void
+    run() override
+    {
+        using ManifestDisposition::accepted;
+        using ManifestDisposition::malformed;
+        using ManifestDisposition::untrusted;
+        using ManifestDisposition::stale;
+        using ManifestDisposition::invalid;
```

Comment:
MSVC doesn't like this syntax / shortcut. I don't know if there's a workaround beyond replacing the abbreviated enums with the fully qualified name.